docs/src/languages/java/migration.md

```diff
@@ -0,0 +1,166 @@
+weight: 2
+sidebarTitle: "Moving to Platform.sh"
+---
+It is common to have a Java application that wants to migrate to Platform.sh. It has support to several styles of Java application such as monolith, microservices, stateful, and stateless.
+## Minimum Requirement
+
+To run a Java application at Platform.sh you must check: 
+
+* [A supported Java version](https://docs.platform.sh/languages/java.html#supported-versions)
+* [A build management](https://docs.platform.sh/languages/java.html#support-build-automation)
+  * [Gradle](https://docs.gradle.org/current/userguide/gradle_wrapper.html)
+  * [Maven](https://maven.apache.org/) 
+  * [Maven Wrapper](https://www.baeldung.com/maven-wrapper)
+  * [Ant](https://ant.apache.org/)
+* A Git Repository:
+  * [GitHub](https://docs.platform.sh/integrations/source/github.html)
+  * [BitBucket](https://docs.platform.sh/integrations/source/bitbucket.html)
+  * [GitLab](https://docs.platform.sh/integrations/source/gitlab.html)
+  * The default Git repository provided by Platform.sh
+
+{{< note >}}
+A container application cannot be bigger than **8 GB**, see  [tunning](https://docs.platform.sh/languages/java/tuning.html) to more details.
+{{< /note >}}
+
+## Monolith/Single Application
+
+To start a Java application, you need to understand the [Platform.sh structure](https://docs.platform.sh/overview/structure.html).  At minimum you will need three [YAML files](https://docs.platform.sh/configuration/yaml.html): 
+
+1. [Application](https://docs.platform.sh/configuration/app.html)
+2. [Route](https://docs.platform.sh/configuration/routes.html)
+
+### Application
+
+```yaml
+name: app
+type: "java:<version>"
+disk: 1024
+hooks:
+    build: 2
+web:
+    commands:
+        start: 3
+```
+
+1. [A Java version](https://docs.platform.sh/languages/java.html#supported-versions), e,g.: `java:11`
+2. [The build defines what happens when building the application](https://docs.platform.sh/configuration/app/build.html#build), This build process might generate an executable file such as a uber-jar e.g.: `mvn clean package`
+3. [The commands key defines the command to launch the application](https://docs.platform.sh/configuration/app/web.html#commands). E.g.:  `java -jar file.jar`
+4. In the start's command needs to receive the port where the application will execute thought the `PORT` environment. That is trivial if your application follows the port bind principle. E.g.: `java -jar jar --port=$PORT`
+
+### Route
+
+```yaml
+"https://{default}/":
+  type: upstream
+  upstream: 1
+"https://www.{default}/":
+  type: redirect
+  to: "https://{default}/"
+```
+
+1. It defines the application will link in the route, e.g.: `"app:http"`
+
+{{< note >}}
+
+Be aware that the container has a maximum memory if the application is stateless to increase a more powerful computer, there is the strategy to use a [load balancer](https://community.platform.sh/t/how-to-configure-load-balancer-in-a-single-application/553).
```

Comment:
I cannot parse this sentence in my head.  What are you saying here?

Reply:
Each container has a maximum limit, we have the option of horizontal scalability with a load balancer.